arch/xtensa/core/cpu_idle.c

```diff
@@ -3,8 +3,15 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <zephyr/toolchain.h>
 #include <zephyr/tracing/tracing.h>
 
+#if defined(__XT_CLANG__) && defined(CONFIG_XTENSA_WAITI_BUG)
+/* xt-clang removes any NOPs more than 8. So we need to set
+ * no optimization to avoid those NOPs from being removed.
+ */
+__no_optimization
```

Comment:
The **C compiler** is peephole optimizing instructions in the middle of the string (!) of an asm() block?  What sort of sorcery is this?  I can only assume it's working by passing some directive through to the assembler?  If so, might be cleaner to use that.

Reply:
Yes... anything NOPs beyond 8 are removed. No idea why...